pointer on interface value

The following program `sample.go` triggers a panic:

```go
package main

import (
    "encoding/xml"
    "fmt"
)

type Email struct {
    Where string `xml:"where,attr"`
    Addr  string
}

func f(s string, r interface{}) {
    err := xml.Unmarshal([]byte(s), &r)
    fmt.Println(err, r)
}

func main() {
    data := `
        <Email where='work'>
            <Addr>bob@work.com</Addr>
        </Email>
    `
    v := Email{}
    f(data, &v)
}
```

Expected result:
```console
$ go run ./sample.go
<nil> &{work bob@work.com}
```

Got:
```console
$ yaegi ./sample.go
./sample.go:14:34: panic
./sample.go:26:2: panic
run: reflect.Set: value of type *interp.valueInterface is not assignable to type *interface {}
goroutine 1 [running]:
runtime/debug.Stack(0x1, 0xc00037c600, 0x40)
        /usr/lib/go/src/runtime/debug/stack.go:24 +0x9f
github.com/traefik/yaegi/interp.(*Interpreter).eval.func1(0xc00012fc70)
        /home/marc/go/src/github.com/traefik/yaegi-master/interp/interp.go:480 +0xc5
panic(0xdcfac0, 0xc00011ee60)
        /usr/lib/go/src/runtime/panic.go:969 +0x1b9
github.com/traefik/yaegi/interp.runCfg.func1(0xc00042e0a0, 0xc000503a00, 0xc00012fa60)
        /home/marc/go/src/github.com/traefik/yaegi-master/interp/run.go:171 +0x247
panic(0xdcfac0, 0xc00011ee60)
        /usr/lib/go/src/runtime/panic.go:969 +0x1b9
github.com/traefik/yaegi/interp.runCfg.func1(0xc00042e140, 0xc000502400, 0xc00012f840)
        /home/marc/go/src/github.com/traefik/yaegi-master/interp/run.go:171 +0x247
panic(0xdcfac0, 0xc00011ee60)
        /usr/lib/go/src/runtime/panic.go:969 +0x1b9
reflect.Value.assignTo(0xdb0f60, 0xc00011d5c0, 0x16, 0xf09b49, 0xb, 0xdb2fe0, 0x0, 0x4d1096, 0xe51d80, 0xdb0f60)
        /usr/lib/go/src/reflect/value.go:2435 +0x405
reflect.Value.Set(0xdb2fe0, 0xc000120290, 0x196, 0xdb0f60, 0xc00011d5c0, 0x16)
        /usr/lib/go/src/reflect/value.go:1564 +0xbd
github.com/traefik/yaegi/interp.addr.func1(0xc00042e140, 0xc0004bef30)
        /home/marc/go/src/github.com/traefik/yaegi-master/interp/run.go:596 +0x13e
github.com/traefik/yaegi/interp.runCfg(0xc000502400, 0xc00042e140)
        /home/marc/go/src/github.com/traefik/yaegi-master/interp/run.go:177 +0x87
github.com/traefik/yaegi/interp.call.func7(0xc00042e0a0, 0xc0003fd6c0)
        /home/marc/go/src/github.com/traefik/yaegi-master/interp/run.go:1104 +0xc85
github.com/traefik/yaegi/interp.runCfg(0xc000503a00, 0xc00042e0a0)
        /home/marc/go/src/github.com/traefik/yaegi-master/interp/run.go:177 +0x87
github.com/traefik/yaegi/interp.(*Interpreter).run(0xc0003fa000, 0xc000503400, 0xc00042e000)
        /home/marc/go/src/github.com/traefik/yaegi-master/interp/run.go:108 +0x2af
github.com/traefik/yaegi/interp.(*Interpreter).eval(0xc0003fa000, 0xc000376480, 0x178, 0x7ffe3a4a7562, 0x15, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /home/marc/go/src/github.com/traefik/yaegi-master/interp/interp.go:572 +0x809
github.com/traefik/yaegi/interp.(*Interpreter).EvalPath(0xc0003fa000, 0x7ffe3a4a7562, 0x15, 0xc000376300, 0x178, 0x0, 0x0, 0xc000136f70)
        /home/marc/go/src/github.com/traefik/yaegi-master/interp/interp.go:394 +0xfa
main.runFile(0xc0003fa000, 0x7ffe3a4a7562, 0x15, 0xc000118240, 0x1)
        /home/marc/go/src/github.com/traefik/yaegi-master/cmd/yaegi/run.go:123 +0xae
main.run(0xc000190030, 0x1, 0x1, 0x13, 0x13)
        /home/marc/go/src/github.com/traefik/yaegi-master/cmd/yaegi/run.go:89 +0x7e5
main.main()
        /home/marc/go/src/github.com/traefik/yaegi-master/cmd/yaegi/yaegi.go:144 +0x415

```

--- a/_test/addr2.go
+++ b/_test/addr2.go
@@ -1,0 +1,29 @@
+package main
+
+import (
+	"encoding/xml"
+	"fmt"
+)
+
+type Email struct {
+	Where string `xml:"where,attr"`
+	Addr  string
+}
+
+func f(s string, r interface{}) error {
+	return xml.Unmarshal([]byte(s), &r)
+}
+
+func main() {
+	data := `
+		<Email where='work'>
+			<Addr>bob@work.com</Addr>
+		</Email>
+	`
+	v := Email{}
+	err := f(data, &v)
+	fmt.Println(err, v)
+}
+
+// Ouput:
+// <nil> {work bob@work.com}

--- a/interp/run.go
+++ b/interp/run.go
@@ -588,12 +588,23 @@ func not(n *node) {
 
 func addr(n *node) {
 	dest := genValue(n)
-	value := genValue(n.child[0])
 	next := getExec(n.tnext)
-
-	n.exec = func(f *frame) bltn {
-		dest(f).Set(value(f).Addr())
-		return next
+	c0 := n.child[0]
+	value := genValue(c0)
+	switch c0.typ.cat {
+	case interfaceT:
+		i := n.findex
+		l := n.level
+		n.exec = func(f *frame) bltn {
+			v := value(f).Interface().(valueInterface).value
+			getFrame(f, l).data[i] = reflect.ValueOf(v.Interface())
+			return next
+		}
+	default:
+		n.exec = func(f *frame) bltn {
+			dest(f).Set(value(f).Addr())
+			return next
+		}
 	}
 }
 

